chimingxuan/client/src/client.cpp

```diff
@@ -0,0 +1,77 @@
+#include "ros/ros.h"
+#include "server/Register.h"
+#include "server/Unregister.h"
+#include "std_msgs/String.h"
+
+#include <sstream>
+
+#include <ctime>
+
+#include <signal.h>
+
+void Unregister(int sig){
+    ros::NodeHandle n;
+    ros::ServiceClient client_unregister = n.serviceClient<server::Unregister>("unregister");
+    server::Unregister unregister_srv;
+    std::string client_name = ros::this_node::getName();
+    unregister_srv.request.name = client_name;
+    if(client_unregister.call(unregister_srv)) {
+        if (unregister_srv.response.success) {
+            ROS_INFO("Unregister success");
+        } else {
+            ROS_INFO("Client %s unregister fail!", client_name.c_str());
+        }
+    } else {
+        ROS_INFO("Failed to call server unregister");
+    }
+
+    ros::shutdown();
+}
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "client");
+    ros::NodeHandle n;
+
+    std::string client_name = ros::this_node::getName();
+    std::string topic = client_name;
+
+    ros::ServiceClient client_register = n.serviceClient<server::Register>("register");
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Lines should be <= 80 characters long  [whitespace/line_length] [2]